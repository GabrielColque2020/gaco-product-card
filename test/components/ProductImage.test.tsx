import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ProductImage, ProductCard } from "../../src/";
import { product2 } from "../data/products";


describe('ProductImage', () => {
	test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
		const rerender = render(<ProductImage img={"https://hola.jpg"}/>);
		expect(rerender.baseElement).toMatchSnapshot();
	});

	test('debe de mostrar el componente con la imagen del producto', () => {
		const rerender = render(
			<ProductCard product={product2}>
				{
					()=> (
						<ProductImage/>
					)
				}
			</ProductCard>
		);
		expect(rerender.baseElement).toMatchSnapshot();
	});
});