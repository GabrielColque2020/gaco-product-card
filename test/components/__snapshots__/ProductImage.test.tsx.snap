// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <img
        alt="Product Image"
        class="productImg undefined"
        src="./coffee-mug.png"
      />
    </div>
  </div>
</body>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<body>
  <div>
    <img
      alt="Product Image"
      class="productImg undefined"
      src="https://hola.jpg"
    />
  </div>
</body>
`;
